packages/@react-spectrum/table/src/TableView.tsx

```diff
@@ -392,8 +392,20 @@ function TableVirtualizer({layout, collection, focusedKey, renderView, renderWra
     headerRef.current.scrollLeft = bodyRef.current.scrollLeft;
   }, [bodyRef]);
 
+  // Track offsetWidth so we have the width regardless of a scrollbar
+  // use that to determine if we should update the tableWidth on visible rect change
+  // if the size change was due to scrollbars appearing, don't bother recalculating
+  // column sizes.
```

Comment:
If the vertical scrollbar appears, wouldn't we have to recalculate the column widths in order to prevent the horizontal scrollbar from appearing?

Reply:
yes, I expect there will be some cases where the table has to scroll a little because we didn't recalculate. I haven't been able to demonstrate it though.

A summary of what is happening:
we calculate column sizes based on available space, but then scroll bars appear, which causes the available space to change, so we calculate column sizes based on the available space, but then scroll bars disappear, which causes the available space to change, so we calculate column sizes based on the available space.

calculating column sizes leads to a state change, which leads to a re-render

My solution is that if the width changed because a scrollbar was introduced/removed, we don't recalculate.

Reply:
Yeah you can see this in the CRUD example, just add enough rows so that it starts vertically scrolling, and the horizontal scrollbar also appears.